x/perp/amm/types/genesis_test.go

```diff
@@ -135,6 +135,76 @@ func TestGenesisState_Validate(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		"overflow": {
+			genesis: &GenesisState{
+				Markets: []Market{
+					{
+						Pair:          asset.MustNewPair("btc:usd"),
+						BaseReserve:   sdk.MustNewDecFromStr("258359429617980926666593621001726127912.880790961315660013"), // will overflow
```

Comment:
Looks good. Could you add a doc comment either here or in the function that explains where the oveflow border is?  It'll be hard for future readers of the code to see that the happy path has its reserves 100 lower than the overflow case.

Reply:
Oops, it had auto-merge on